### PR TITLE
fix: add init function for staleStakesForbidden in BLSSignatureChecker

### PR DESCRIPTION
**Pull request by stevennevins**

### CODE REVIEW

File: src/BLSSignatureChecker.sol [hunk 1]
```diff
@@ -15,13 +17,13 @@ import {BN254} from "./libraries/BN254.sol";
  * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
  * @notice This is the contract for checking the validity of aggregate operator signatures.
  */
-contract BLSSignatureChecker is IBLSSignatureChecker {
+contract BLSSignatureChecker is Initializable, IBLSSignatureChecker {
```
Comment (ChaoticWalrus) [inline src/BLSSignatureChecker.sol]: I am somewhat worried about the storage layout change here.
Can we do without this import? I think the only way this storage layout is backwards-compatible is if you're upgrading a contract that inherits from Initializable directly before inheriting from this contract? Seems like a large potential footgun to get an initializer for a single boolean

**Review by ChaoticWalrus — CHANGES_REQUESTED** [review]
The idea is good to me, but I think the storage layout change is likely worth blocking this on.

File: src/BLSSignatureChecker.sol (hunk 1 above)
Comment (stevennevins) [reply src/BLSSignatureChecker.sol]: makes sense, anything that is also initializable shouldn't have shift, but i understand the concern

Comment (ChaoticWalrus) [reply src/BLSSignatureChecker.sol]: the linearization of the inheritance is ultimately what matters, so just inheriting Initializable from elsewhere isn't a guarantee either 😞

**Review by 0x0aa0 — APPROVED** [review]
Am much happier without the initializer here as it requires no change to DASM. Seems like we just need to communicate to AVS teams that this is something they need to flip the switch on some time after running AVSSync

**Review by ChaoticWalrus — APPROVED** [review]
Works for me now.
Thanks for accommodating my request here.

File: src/BLSSignatureChecker.sol (hunk 1 above)
Comment (stevennevins) [reply src/BLSSignatureChecker.sol]: yeah that's true, seems like it could be a foot gun

**Pull request merged**